java/ql/src/experimental/Security/CWE/CWE-598/SensitiveGetQuery.ql

```diff
@@ -0,0 +1,58 @@
+/**
+ * @name Sensitive GET Query
+ * @description Use of GET request method with sensitive query strings.
+ * @kind path-problem
+ * @id java/sensitive-query-with-get
+ * @tags security
+ *       external/cwe-598
+ */
+
+import java
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.dataflow.TaintTracking
+import semmle.code.java.frameworks.Servlets
+import semmle.code.java.security.SensitiveActions
+import DataFlow::PathGraph
+
+/** Finds variables that hold sensitive information judging by their names. */
+class SensitiveInfoExpr extends Expr {
+  SensitiveInfoExpr() {
+    exists(Variable v | this = v.getAnAccess() |
+      v.getName().regexpMatch(getCommonSensitiveInfoRegex())
+    )
+  }
+}
+
+/** GET servlet method of `javax.servlet.http.Servlet` and subtypes. */
+private predicate isGetServletMethod(Callable c) {
+  c.getDeclaringType() instanceof ServletClass and
+  c.getNumberOfParameters() = 2 and
+  c.getParameter(1).getType() instanceof ServletResponse and
```

Comment:
Why check the type of one but not the other parameter?

Reply:
Since <code>ServletRequest</code> is a library class, it can only be used in another library class. I agree we shall specify type of the first parameter to make the query more clear. I've factored   <code>isGetServletMethod</code> into a common method of the <code>java/ql/src/semmle/code/java/frameworks/Servlets.qll</code> library.